src/Components/Web/src/Virtualization/InfiniteScroll.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Rendering;
+
+namespace Microsoft.AspNetCore.Components.Virtualization
+{
+    /// <summary>
+    /// Provides functionality for rendering a virtualized list from a source of unknown size.
+    /// </summary>
+    /// <typeparam name="TItem"></typeparam>
+    public sealed class InfiniteScroll<TItem> : ComponentBase
+    {
+        /// <summary>
+        /// Gets or sets the item template for the list.
+        /// </summary>
+        [Parameter]
+        public RenderFragment<TItem> Item { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets the template for the footer of the list.
+        /// </summary>
+        [Parameter]
+        public RenderFragment<int> Footer { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets the function retrieving items given a start index.
+        /// </summary>
+        [Parameter]
+        public ItemsProviderDelegate<TItem> ItemsProvider { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets the item count requested to the <see cref="ItemsProvider"/>.
+        /// </summary>
+        [Parameter]
+        public int RequestedItemCount { get; set; } = 50;
```

Comment:
It's tricky to know what a sensible default is here, since it's scenario-specific.

Maybe we should err on the side of making it smaller (e.g., 10) so the developer is forced to increase it when it makes sense in their application. Otherwise it's possible they will never realise they are loading chunks of 50 when maybe that's wildly excessive in their scenario (e.g., a feed of complete newspaper articles, not just headlines).

Not a big deal of course!

Reply:
I've changed it to 10 😃